dipy/core/geometry.py

```diff
@@ -125,8 +125,11 @@ def cart2sphere(x, y, z):
        azimuth angle
     '''
     r = np.sqrt(x * x + y * y + z * z)
+    # r is strictly positive by construction and will be 0 iff all three are
+    # zero, so it's safe to convert it to a 1 in this case, and prevents
+    # warnings from being raised in the following statement (z / r):
+    r = np.where(r > 0, r, 1)
     theta = np.arccos(z / r)
-    theta = np.where(r > 0, theta, 0.)
```

Comment:
Since Travis is not happy because of some round values, why not just put it all in one line?  

`theta = np.where(r > 0, np.arccos(z / r), 0.) `

Reply:
or even better : `theta = np.arccos(z / r, where=r>0)`

cool feature from Numpy! look [here](https://stackoverflow.com/a/37977222)